test/ProfilerTest.cpp

```diff
@@ -1,8 +1,98 @@
+#include <arpa/inet.h>
+
+#include <thread>
+
 #include "gtest/gtest.h"
+#include "pipert/Scheduler.h"
+
+namespace {
+const std::uint16_t server_port = 8888;
+const int buffer_size = 131;
+std::uint8_t buffer[buffer_size];
+int sockfd;
+
+void PrepareToReceive() {
+  struct sockaddr_in servaddr;
+
+  // Creating socket file descriptor
+  if ((sockfd = socket(AF_INET, SOCK_DGRAM, 0)) < 0) {
+    ASSERT_TRUE(false);  // socket creation failed
+  }
+
+  memset(&servaddr, 0, sizeof(servaddr));
+
+  // Filling server information
+  servaddr.sin_family = AF_INET;  // IPv4
+  servaddr.sin_addr.s_addr = inet_addr("127.0.0.1");
+  servaddr.sin_port = htons(server_port);
+
+  // Bind the socket with the server address
+  if (bind(sockfd, (const struct sockaddr *)&servaddr, sizeof(servaddr)) < 0) {
+    ASSERT_TRUE(false);  // bind failed
+  }
+}
+
+bool Receive() {
+  struct sockaddr_in cliaddr;
+  memset(&cliaddr, 0, sizeof(cliaddr));
+  socklen_t len = sizeof(cliaddr);
+  int retbuflen = recvfrom(sockfd, buffer, buffer_size, 0,
+                           (struct sockaddr *)&cliaddr, &len);
+  if (retbuflen != buffer_size) return false;
+  if (buffer[0] == 68U && buffer[37] == 71U && buffer[94] == 115U) {
+    close(sockfd);
+    return true;
+  } else {
+    close(sockfd);
+    return false;
+  }
+}
+
+class Human {
```

Comment:
These classes are copy-pasted too much in our tests. It would be good to factor these out into one single header from where we can include.